### DISCUSSION TITLE
Is there any plan update to electron12?

### DISCUSSION THREAD

**Discussion by linhanyu** [eclipse-theia/theia#9854] [Q&A] [answered]
accroding to https://github.com/eclipse-theia/theia/pull/9813

I have compile project in my m1 laptop, And there is two probleam:

1. the Nodejs version of electron 11 were 12.x. It's not good support in M1-laptop. It make compile very slow, about 5 minutes in my x86_64 laptop and 2 hours in m1-laptop

2. Also, ffmepg replace were fail because of the architecture (the ffmepeg.node were x86_64, any plan to replace?)

also electron11 concern m1 support, but I think it's hard to debug with node12 and may cause another probleam.

Many users face crash issue when we use rosetta2 to run x86_64 application. 
https://github.com/eclipse-theia/theia/issues/9846

**It's main resistance to promote our production.**
I realizeelectron12 update nodejs to 14. it faster than 12 in m1-devices, but many dependency must be update (such as find-git).I think it's hard for me to upgrade all of them.

**Comment by vince-fugnitto** [comment, marked as answer]
@linhanyu I do believe we should try to aim towards newer versions of `electron` and especially node (so we target the LTS).
It will likely be a long process:
- ensuring that we can build with newer versions of node, update our dependencies
- perform the electron license compatibility check by the eclipse foundation (which may take a while)
- confirm that proprietary codecs are not used (reason for the `ffmpeg` replacement)

In addition, vscode also targets newer electron and node versions and it would be good to align :+1:

**Reply by luckymark** [reply]
any update? electron 9 is really old version!

thanks for your great work!

**Reply by msujew** [reply]
@luckymark We are currently in the process of updating Electron. See https://github.com/eclipse-theia/theia/pull/9936. We are currently slowed down by the Eclipse Foundation's CQ process.

**Reply by luckymark** [reply]
thanks a lot
